pkg/alertmanager/state_replication.go

```diff
@@ -41,6 +47,14 @@ type state struct {
 // newReplicatedStates creates a new state struct, which manages state to be replicated between alertmanagers.
 func newReplicatedStates(userID string, rf int, re Replicator, l log.Logger, r prometheus.Registerer) *state {
 
+	defaultSettleConfig := util.BackoffConfig{
```

Comment:
I think this config would end up retrying for too long (5 retries, 10s max backoff, 15s timeout). Before discussing how to fine tune it, question is: why do we need retries at all?

I understand "everything fail all the time", but do we need retries also for the normal use case (eg. the other replica is not "ready" yet) or just on case of failures while fetching it? If retries are to cover the latter case, then maybe we don't need retries at all, considering the design is to fallback reading state from the storage (which will be one of the next things we'll build).

Reply:
I don't really have a good answer for this, other than _that was the plan_. Maybe @gotjosh has an opinion?

I suppose we want to try and cover transient errors as best as possible? But given we're using a more reliable means of fetching the state, which itself has a timeout, and as you say, we have another fallback to S3, maybe we can take the retries away entirely.